src/plots.jl

```diff
@@ -337,8 +337,17 @@ function Histogram2(
         scale =  xbins * ybins / ((xmax-xmin) * (ymax-ymin) * sum(M))
         M = M * scale
     end
-    if zmode == "log" 
-        M = M .+ 1
+    if zmode == "log"
+        if zmin == nothing
+            if minimum(M) == 0
+                nonzeromin = minimum(M[findall(x->x!=0, M)])
+                M = replace!(Float64.(M), 0. =>nonzeromin/2)
+            end
+        elseif zmin <= 0
+            error("zmin must be > 0 for a valid log10 output")
+        else
+            M = replace!(Float64.(M), 0., zmin)
+        end
         M = log10.(M)
         return Image(M, (xmin, xmax), (ymin, ymax), filename=filename, colorbar=colorbar, colorbarStyle="yticklabel=\$10^{\\pgfmathprintnumber{\\tick}}\$", colormap=colormap, zmin=zmin, zmax=zmax, style=style)
```

Comment:
I thought we were going to switch to `zmode=log`?

Reply:
we are using zmode and zmin right? zmode sets it to log earlier.

Reply:
I'm talking about the colorbarStyle.